goss-testing/suites/ncn-storage-tests.yaml

```diff
@@ -29,6 +29,7 @@
 {{ $thisnode := .Vars.thisnode }}
 gossfile:
     ../tests/goss-ceph-csi-storage-node-requirements.yaml: {}
+    ../tests/goss-ncn-xname.yaml: {}
```

Comment:
Do we have to check this here if we are already checking it in ncn-healthcheck-storage.yaml?

Reply:
I think we should. During the install, after deploying management nodes, we run "csi pit validate --ceph" and "csi pit validate --k8s". These run automated/ncn-storage-checks and automated/ncn-kubernetes-checks respectively.

**automated/ncn-storage-checks**
Prior to my changes, this ran suites/ncn-storage-tests.yaml on ncn-s001. After my changes, it runs that suite on all of the storage nodes.

**automated/ncn-kubernetes-checks**
This runs suites/ncn-kubernetes-tests-master.yaml on all of the masters and suites/ncn-kubernetes-tests-worker.yaml on all of the workers. (When this script is run post-PIT-reboot, it also runs the ncn-afterpitreboot- versions of these suites on these same nodes).

The suite you called out (ncn-healthcheck-storage.yaml) only runs during the regular CSM healthcheck (whereas the ncn-storage-tests suite does NOT run at that point). 

But in answering you I realized I do need to make a change to the PR -- I should be adding the xname test to the ncn-kubernetes-tests suite for masters and workers, rather than in the ncn-healthcheck suites. The kubernetes-tests run both after management note deploy and during regular CSM healthcheck, so it's a better place for the xname test.

Reply:
I think you still want to add it to ncn-healthcheck-worker and ncn-healthcheck-master.   I don't see ncn-kubernetes-tests-master or ncn-kubernetes-tests-worker called during healthchecks.

```
ncn-m001:/opt/cray/tests/install # grep -r ncn-kubernetes-tests-master *
livecd/automated/ncn-kubernetes-checks:    run_ncn_tests "$node" "8996" "ncn-kubernetes-tests-master" ; \
livecd/automated/ncn-kubernetes-checks:    run_ncn_tests "$node" "8996" "ncn-kubernetes-tests-master" ; \
ncn/automated/ncn-kubernetes-checks:    run_ncn_tests "$node" "8996" "ncn-kubernetes-tests-master" ; \
ncn/automated/ncn-kubernetes-checks:    run_ncn_tests "$node" "8996" "ncn-kubernetes-tests-master" ; \
```

```
ncn-m001:/opt/cray/tests/install # grep -r ncn-kubernetes-tests-worker *
livecd/automated/ncn-kubernetes-checks:    run_ncn_tests "$node" "8998" "ncn-kubernetes-tests-worker"; \
livecd/automated/ncn-kubernetes-checks:    run_ncn_tests "$node" "8998" "ncn-kubernetes-tests-worker" ; \
ncn/automated/ncn-kubernetes-checks:    run_ncn_tests "$node" "8998" "ncn-kubernetes-tests-worker"; \
ncn/automated/ncn-kubernetes-checks:    run_ncn_tests "$node" "8998" "ncn-kubernetes-tests-worker" ; \
```

Reply:
Also, the more I work on these tickets, the more I regret some of our test suite name choices. :)